packages/dashboard/src/components/widgets/primitives/input/index.tsx

```diff
@@ -1,22 +1,32 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
 import { Button, Select, SelectProps } from '@cloudscape-design/components';
 import SpaceBetween from '@cloudscape-design/components/space-between';
 import { InputWidget as InputWidgetType } from '../../../../types';
 
 export type InputWidgetProps = InputWidgetType & { readOnly: boolean };
 
 const Input: React.FC<InputWidgetProps> = ({ readOnly, ...widget }) => {
-  const [selectedOption, setSelectedOption] = useState<SelectProps.Option>(widget.options[0]);
+  const [selectedOption, setSelectedOption] = useState<SelectProps.Option | null>(widget.options[0]);
+  const options = widget.options.map(({ label }) => ({ label, value: label }));
+  const disabled = !readOnly || options.length === 0;
+
+  const changeOption: NonCancelableEventHandler<SelectProps.ChangeDetail> = ({ detail: { selectedOption } }) => {
+    setSelectedOption(selectedOption);
+  };
+
+  useEffect(() => {
+    if (options.length === 0) {
+      setSelectedOption(null);
+    } else if (!selectedOption || options.filter(({ label }) => selectedOption.label === label).length === 0) {
+      setSelectedOption(widget.options[0]);
+    }
+  }, [widget.options]);
 
   return (
     <SpaceBetween size={'xs'} direction={'horizontal'}>
-      <Select
-        disabled={!readOnly}
-        selectedOption={selectedOption}
-        onChange={({ detail }) => setSelectedOption(detail.selectedOption)}
-        options={widget.options}
-      />
-      <Button disabled={!readOnly}>{widget.messageOverrides?.submitLabel}</Button>
+      <Select disabled={disabled} selectedOption={selectedOption} onChange={changeOption} options={options} />
+      <Button disabled={disabled}>{widget.messageOverrides?.submitLabel}</Button>
```

Comment:
should submitLabel really sometime be undefined? feels like it should be required after merging in defaults.

Reply:
Agreed, this should not be optional. This is the pattern we have been following for all the widgets so far though. Please look at `AppKitWidget`, `TextWidget` etc. I will make it required for my component but we should fix this for the other widgets as well. I won't include that in my PR but worth tracking.

Reply:
We should be deep merging message overrides at the dashboard component so you shouldn't need to worry about having undefined messaging.

Reply:
we should fix it in this PR and start enforcing better patterns around message over rides

Reply:
Looks like this won't be a simple fix since this is broken all over the dashboard.